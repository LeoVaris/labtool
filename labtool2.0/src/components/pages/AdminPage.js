import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Button, Loader, Label, Table, Popup } from 'semantic-ui-react'
import { Link } from 'react-router-dom'
import { Redirect } from 'react-router'
import { HorizontalScrollable } from '../HorizontalScrollable'
import { formatCourseName } from '../../util/format'
import { getAllCI } from '../../services/courseInstance'
import { resetLoading } from '../../reducers/loadingReducer'
import { getAllUsers, updateOtherUser } from '../../services/user'
import { clearNotifications } from '../../reducers/notificationReducer'
import { usePersistedState } from '../../hooks/persistedState'
import { sortUsersBySysopLastname } from '../../util/sort'

const AdminCourseList = ({ courses }) => {
  if (!Array.isArray(courses)) {
    return <div />
  }

  return (
    <HorizontalScrollable>
      <Table singleLine color="yellow" style={{ overflowX: 'visible' }}>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell colSpan="1">Course ID</Table.HeaderCell>
            <Table.HeaderCell colSpan="1">Course name</Table.HeaderCell>
            <Table.HeaderCell colSpan="1">Course start date</Table.HeaderCell>
            <Table.HeaderCell colSpan="2"> </Table.HeaderCell>
          </Table.Row>
        </Table.Header>

        <Table.Body>
          {courses.map(instance => (
            <Table.Row key={instance.id}>
              <Table.Cell>{instance.shorterId} </Table.Cell>
              <Table.Cell>
                <strong>{formatCourseName(instance.name, instance.ohid, instance.start)}</strong>
              </Table.Cell>

              <Table.Cell> {instance.europeanStart} </Table.Cell>
              <Table.Cell textAlign="center">
                <Popup
                  trigger={
                    <Button
                      circular
                      size="tiny"
                      icon={{ name: 'edit', size: 'large', color: 'orange' }}
                      as={Link}
                      to={{
                        pathname: `/labtool/ModifyCourseInstanceStaff/${instance.ohid}`,
                        state: { fromAdmin: true }
                      }}
                    />
                  }
                  position="top right"
                  content="Manage instructors for this course"
                />
              </Table.Cell>
            </Table.Row>
          ))}
        </Table.Body>
      </Table>
    </HorizontalScrollable>
  )
}

AdminCourseList.propTypes = {
  courses: PropTypes.oneOfType([PropTypes.array, PropTypes.object])
}

const AdminUserList = ({ me, users, updateOtherUser }) => {
  const setAdmin = (userId, name, newAdmin) => () => {
    const message = `Are you sure you want to ${newAdmin ? 'grant admin rights to' : 'remove admin rights from'} ${name}?`
    if (window.confirm(message)) {
      const selfInflicted = userId === me.user.id
      if (selfInflicted && !newAdmin && !window.confirm('You probably do NOT want to remove admin rights from yourself. Are you really sure?')) {
        return
      }
      updateOtherUser({ id: userId, sysop: newAdmin })
    }
  }
  const sortedUsers = sortUsersBySysopLastname(users)

  return (
    <Table singleLine color="yellow" fixed>
      <Table.Header>
        <Table.Row>
          <Table.HeaderCell>Name</Table.HeaderCell>
          <Table.HeaderCell>Status</Table.HeaderCell>
        </Table.Row>
      </Table.Header>
      <Table.Body>
        {sortedUsers.map(user => (
          <Table.Row key={user.id}>
            <Table.Cell>
              {user.firsts} {user.lastname}
              <br />
              <a href={`mailto:${user.email}`}>{user.email}</a>
            </Table.Cell>
            <Table.Cell>
              {user.teacher && (
                <Label color="orange" horizontal>
                  Teacher
                </Label>
              )}
              {user.sysop ? (
                <div>
                  <Label color="orange" horizontal>
                    Admin
                  </Label>
                  <Button onClick={setAdmin(user.id, `${user.firsts} ${user.lastname}`, false)} size="tiny" color="green">
                    Remove admin
                  </Button>
                </div>
              ) : (
                <div>
                  <Label horizontal>Not admin</Label>
                  <Button onClick={setAdmin(user.id, `${user.firsts} ${user.lastname}`, true)} size="tiny" color="green">
                    Make admin
                  </Button>
                </div>
              )}
            </Table.Cell>
          </Table.Row>
        ))}
      </Table.Body>
    </Table>
  )
}

AdminUserList.propTypes = {
  me: PropTypes.object.isRequired,
  users: PropTypes.array.isRequired,
  updateOtherUser: PropTypes.func.isRequired
}

export const AdminPage = props => {
  const state = usePersistedState('AdminPage', {
    coursesList: false,
    usersList: false
  })

  useEffect(() => {
    props.clearNotifications()
  }, [])

  useEffect(() => {
    if (state.coursesList) {
      props.resetLoading()
      props.getAllCI()
    }
  }, [state.coursesList])

  useEffect(() => {
    if (state.usersList) {
      props.resetLoading()
      props.getAllUsers()
    }
  }, [state.usersList])

  if (!props.user.user.sysop) {
    return <Redirect to={`/labtool`} />
  } else if (props.loading.loading) {
    return <Loader active />
  }

  return (
    <div className="AdminPage">
      <h2>Welcome to the admin interface.</h2>
      <Button compact className={`ui button`} onClick={() => (state.coursesList = !state.coursesList)}>
        {state.coursesList ? 'Hide courses' : 'Manage courses'}
      </Button>
      <br />
      <br />
      {state.coursesList && <AdminCourseList courses={props.courseInstance} />}
      <br />
      <hr />
      <br />
      <Button compact className={`ui button`} onClick={() => (state.usersList = !state.usersList)}>
        {state.usersList ? 'Hide users' : 'Manage users'}
      </Button>
      <br />
      {state.usersList && <AdminUserList me={props.user} users={props.users} updateOtherUser={props.updateOtherUser} />}
    </div>
  )
}

const mapStateToProps = state => {
  return {
    user: state.user,
    courseInstance: state.courseInstance,
    users: state.users,
    loading: state.loading
  }
}

const mapDispatchToProps = {
  getAllCI,
  resetLoading,
  getAllUsers,
  updateOtherUser,
  clearNotifications
}

AdminPage.propTypes = {
  user: PropTypes.object.isRequired,
  courseInstance: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
  users: PropTypes.array.isRequired,
  loading: PropTypes.object.isRequired,

  getAllCI: PropTypes.func.isRequired,
  resetLoading: PropTypes.func.isRequired,
  getAllUsers: PropTypes.func.isRequired,
  updateOtherUser: PropTypes.func.isRequired,
  clearNotifications: PropTypes.func.isRequired
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AdminPage)
