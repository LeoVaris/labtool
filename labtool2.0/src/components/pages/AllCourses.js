import React, { Component } from 'react'
import { getAll } from '../../services/courseInstance'
import { Link } from 'react-router-dom'

class AllCourses extends Component {
  constructor(props) {
    super(props)
    this.state = {
      courses: [],
    }
  }

  componentDidMount() {
    getAll().then(courses =>
      this.setState({ courses })
    )
  }

  render() {
    return (
      <div>
        <ul>
          {
            this.state.courses.map(c => (
              <li key={c.id}>
                <Link to={`/courses/${c.id}`}>{c.name}</Link>
              </li>
            ))
          }
        </ul>
      </div>
    )
  }
}



export default AllCourses