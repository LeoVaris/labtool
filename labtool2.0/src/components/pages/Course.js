import React, { Component } from 'react'
import { getOne } from '../../services/courseInstance'



class Course extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      course: undefined,
    }
  }

  componentDidMount() {
    getOne(parseInt(this.props.match.params.number, 10)).then(
      course => this.setState({ course })
    )
  }

  render() {
    if (!this.state.course) {
      return <div>course not found!!!</div>
    }

    return (
      <div>
        <h1>{this.state.course.name} </h1>
      </div>
    )
  }
}

export default Course