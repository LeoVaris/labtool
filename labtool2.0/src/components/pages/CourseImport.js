import React, { Component } from 'react'
import { Button, Container, Header, Table, Loader, Form } from 'semantic-ui-react'
import { connect } from 'react-redux'
import { Link, Redirect } from 'react-router-dom'
import { getIsAllowedToImport, getImportableCourses, importCourses } from '../../services/courseImport'
import { resetLoading, addRedirectHook } from '../../reducers/loadingReducer'
import { HorizontalScrollable } from '../HorizontalScrollable'

/**
 *  Show all the courses in a single list.
 */
export class CourseImport extends Component {
  componentWillMount = async () => {
    await this.props.resetLoading()
    this.props.getIsAllowedToImport()
    this.props.getImportableCourses()
  }

  handleSubmit = async e => {
    e.preventDefault()
    const data = this.props.importable
    if (data) {
      const courses = data.filter(entry => e.target['import' + entry.hid] && e.target['import' + entry.hid].checked)
      this.props.addRedirectHook({
        hook: 'COURSE_IMPORT_DO_IMPORT_'
      })
      await this.props.importCourses({ courses })
    }
  }

  render() {
    return (
      <div className="CourseImport">
        <Container>
          <Header as="h2">Import courses</Header>
          {this.props.loading.loading || !Array.isArray(this.props.importable) || !this.props.canImport ? (
            <Loader active />
          ) : this.props.loading.redirect ? (
            <Redirect to="/labtool/courses" />
          ) : this.props.importable.length ? (
            <Form onSubmit={this.handleSubmit}>
              <HorizontalScrollable>
                <Table singleLine color="yellow" style={{ overflowX: 'visible' }}>
                  <Table.Header>
                    <Table.Row>
                      <Table.HeaderCell colSpan="1" key={-1}>
                        Import?
                      </Table.HeaderCell>
                      <Table.HeaderCell colSpan="1">Course id</Table.HeaderCell>
                      <Table.HeaderCell colSpan="1">Course name</Table.HeaderCell>
                      <Table.HeaderCell colSpan="1">Course start date</Table.HeaderCell>
                      <Table.HeaderCell colSpan="1">Course end date</Table.HeaderCell>
                    </Table.Row>
                  </Table.Header>

                  <Table.Body>
                    {this.props.importable.map(instance => (
                      <Table.Row key={instance.hid}>
                        <Table.Cell>
                          <Form.Checkbox name={'import' + instance.hid} />
                        </Table.Cell>
                        <Table.Cell>{instance.shorterId}</Table.Cell>
                        <Table.Cell>{instance.cname}</Table.Cell>
                        <Table.Cell>{instance.europeanStart}</Table.Cell>
                        <Table.Cell>{instance.europeanEnd}</Table.Cell>
                      </Table.Row>
                    ))}
                  </Table.Body>
                </Table>
              </HorizontalScrollable>

              <br />
              <br />

              <Button type="submit" className="ui green button" content="Import" />

              <br />
              <br />

              <Link to="/labtool/courses">
                <Button className="ui button" type="cancel">
                  Cancel
                </Button>
              </Link>
            </Form>
          ) : (
            <div>
              <h4>There are no courses in Kurki to import</h4>

              <Link to="/labtool/courses">
                <Button className="ui button" type="cancel">
                  Back
                </Button>
              </Link>
            </div>
          )}
        </Container>
      </div>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    loading: state.loading,
    canImport: state.courseImport.canImport,
    importable: state.courseImport.importable
  }
}

const mapDispatchToProps = {
  resetLoading,
  getIsAllowedToImport,
  getImportableCourses,
  importCourses,
  addRedirectHook
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CourseImport)
