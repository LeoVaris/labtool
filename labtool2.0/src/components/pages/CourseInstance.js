import React from 'react'

const CourseInstance = ({ instance, handleFieldChange }) => {
  return(
    <li style={{ textAlignVertical: 'center', textAlign: 'center' }}> {instance.name} <button name={instance.name} onClick={handleFieldChange} value={instance.id}>Register</button>
    </li>
  )
}

export default CourseInstance