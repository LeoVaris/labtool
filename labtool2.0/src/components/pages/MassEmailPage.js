import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import { Button, Form, Header, Loader } from 'semantic-ui-react'
import { Link, Redirect } from 'react-router-dom'
import { connect } from 'react-redux'
import { getOneCI, coursePageInformation } from '../../services/courseInstance'
import { getAllTags } from '../../services/tags'
import { sendMassEmail } from '../../services/email'
import { coursePageReset, restoreStudentSelection } from '../../reducers/coursePageLogicReducer'
import { resetLoading, addRedirectHook } from '../../reducers/loadingReducer'
import store from '../../store'
import StudentTable from '../StudentTable'

export const MassEmailPage = props => {
  useEffect(() => {
    // run on component mount
    props.resetLoading()
    props.getOneCI(props.courseId)
    props.coursePageInformation(props.courseId)
    props.getAllTags()

    // do not carry over selection changes back to main student table
    const oldSelectedStudents = { ...props.coursePageLogic.selectedStudents }
    return () => {
      // run on component unmount
      props.coursePageReset()
      props.restoreStudentSelection(oldSelectedStudents)
    }
  }, [])

  const handleSubmit = async e => {
    e.preventDefault()
    const data = props.courseData && props.courseData.data
    if (data) {
      const sendingTo = data.filter(entry => props.coursePageLogic.selectedStudents[entry.id]).map(entry => ({ id: entry.id }))

      if (!sendingTo.length) {
        store.dispatch({ type: 'MASS_EMAIL_SENDFAILURE' })
      } else {
        props.addRedirectHook({
          hook: 'MASS_EMAIL_SEND'
        })
        await props.sendMassEmail({ students: sendingTo, content: e.target.content.value, sendToInstructors: e.target.sendToInstructors.checked }, props.selectedInstance.ohid)
      }
    }
  }

  if (props.loading.loading) {
    return <Loader active />
  }
  if (props.loading.redirect) {
    return <Redirect to={`/labtool/courses/${props.selectedInstance.ohid}`} />
  }

  /**
   * Function that returns what teachers should see at this page
   * We will use a form to decide which students to send an email
   * and what exactly to send
   */
  let renderTeacherPart = () => {
    return (
      <div className="TeacherMassEmailPart">
        <Header as="h2">Send email to students </Header>

        <Form onSubmit={handleSubmit}>
          <StudentTable
            rowClassName="TableRowForActiveStudents"
            columns={['select']}
            allowModify={false}
            filterStudents={data => data.User.email}
            disableDefaultFilter={false}
            selectedInstance={props.selectedInstance}
            courseData={props.courseData}
            coursePageLogic={props.coursePageLogic}
            tags={props.tags}
          />

          <br />
          <br />

          <Form.TextArea name="content" placeholder="Type email here..." defaultValue="" required />
          <Form.Checkbox name="sendToInstructors" defaultChecked={true} label="Send a copy to all instructors" />
          <Button type="submit" className="ui green button" content="Send" />

          <br />
          <br />

          <Link to={`/labtool/courses/${props.selectedInstance.ohid}`}>
            <Button className="ui button" type="cancel">
              Cancel
            </Button>
          </Link>
        </Form>
        <br />
      </div>
    )
  }

  /**
   * This part actually tells what to show to the user
   */
  if (props.courseData.role === 'teacher') {
    return (
      <div style={{ overflowX: 'auto', overflowY: 'hidden' }}>
        {renderTeacherPart()}
        <br />
        <br />
        <br />
        <br />
        <br />
        <br />
        <br />
      </div>
    )
  } else {
    return <div />
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    user: state.user,
    studentInstance: state.studentInstance,
    teacherInstance: state.teacherInstance,
    selectedInstance: state.selectedInstance,
    courseInstance: state.courseInstance,
    courseData: state.coursePage,
    coursePageLogic: state.coursePageLogic,
    courseId: ownProps.courseId,
    tags: state.tags,
    loading: state.loading
  }
}

const mapDispatchToProps = {
  getOneCI,
  coursePageInformation,
  coursePageReset,
  getAllTags,
  sendMassEmail,
  resetLoading,
  addRedirectHook,
  restoreStudentSelection
}

MassEmailPage.propTypes = {
  courseId: PropTypes.string.isRequired,

  user: PropTypes.object.isRequired,
  studentInstance: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
  teacherInstance: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
  selectedInstance: PropTypes.object.isRequired,
  courseInstance: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
  courseData: PropTypes.object.isRequired,
  coursePageLogic: PropTypes.object.isRequired,
  tags: PropTypes.object.isRequired,
  loading: PropTypes.object.isRequired,

  getOneCI: PropTypes.func.isRequired,
  coursePageInformation: PropTypes.func.isRequired,
  coursePageReset: PropTypes.func.isRequired,
  getAllTags: PropTypes.func.isRequired,
  sendMassEmail: PropTypes.func.isRequired,
  resetLoading: PropTypes.func.isRequired,
  addRedirectHook: PropTypes.func.isRequired,
  restoreStudentSelection: PropTypes.func.isRequired
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MassEmailPage)
