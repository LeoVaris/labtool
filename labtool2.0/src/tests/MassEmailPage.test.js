import React from 'react'
import { MassEmailPage } from '../components/pages/MassEmailPage'
import { shallow } from 'enzyme'

describe('<MassEmailPage />', () => {
  let wrapper

  const coursePage = {
    role: 'teacher',
    data: [
      {
        id: 10012,
        github: 'http://github.com/tiralabra2',
        projectName: 'Tiran toinen labraprojekti',
        createdAt: '2018-03-26T00:00:00.000Z',
        updatedAt: '2018-03-26T00:00:00.000Z',
        courseInstanceId: 10011,
        userId: 10012,
        teacherInstanceId: 10011,
        weeks: [],
        codeReviews: [],
        User: {
          id: 10012,
          username: 'tiraopiskelija2',
          email: 'johan.studerande@helsinki.fi',
          firsts: 'Johan Wilhelm',
          lastname: 'Studerande',
          studentNumber: '014553242',
          admin: false,
          createdAt: '2018-03-26T00:00:00.000Z',
          updatedAt: '2018-03-26T00:00:00.000Z'
        },
        Tags: [
          {
            id: 20001,
            name: 'Javascript',
            color: 'red'
          }
        ]
      },
      {
        id: 10031,
        github: 'http://github.com/superprojekti',
        projectName: 'Tira super projekti',
        createdAt: '2018-03-26T00:00:00.000Z',
        updatedAt: '2018-06-05T07:12:28.603Z',
        courseInstanceId: 10011,
        userId: 10031,
        teacherInstanceId: 10011,
        weeks: [],
        codeReviews: [],
        User: {
          id: 10031,
          username: 'superopiskelija',
          email: 'teras.henkilo@helsinki.fi',
          firsts: 'Teräs',
          lastname: 'Henkilö',
          studentNumber: '014666666',
          admin: false,
          createdAt: '2018-03-26T00:00:00.000Z',
          updatedAt: '2018-03-26T00:00:00.000Z'
        },
        Tags: [
          {
            id: 20008,
            name: 'DROPPED',
            color: 'grey'
          },
          {
            id: 20002,
            name: 'HTML',
            color: 'yellow'
          }
        ]
      },
      {
        id: 10011,
        github: 'http://github.com/tiralabra1',
        projectName: 'Tiran labraprojekti',
        createdAt: '2018-03-26T00:00:00.000Z',
        updatedAt: '2018-03-26T00:00:00.000Z',
        courseInstanceId: 10011,
        userId: 10011,
        teacherInstanceId: 10011,
        weeks: [],
        codeReviews: [],
        User: {
          id: 10011,
          username: 'tiraopiskelija1',
          email: 'maarit.opiskelija@helsinki.fi',
          firsts: 'Maarit Mirja',
          lastname: 'Opiskelija',
          studentNumber: '014578343',
          admin: false,
          createdAt: '2018-03-26T00:00:00.000Z',
          updatedAt: '2018-03-26T00:00:00.000Z'
        },
        Tags: []
      }
    ]
  }

  const tags = {
    tags: [
      {
        id: 20001,
        name: 'Javascript',
        color: 'red',
        createdAt: '2018-06-13T00:00:00.000Z',
        updatedAt: '2018-06-13T00:00:00.000Z'
      },
      {
        id: 20002,
        name: 'HTML',
        color: 'yellow',
        createdAt: '2018-06-13T00:00:00.000Z',
        updatedAt: '2018-06-13T00:00:00.000Z'
      },
      {
        id: 20003,
        name: 'game',
        color: 'black',
        createdAt: '2018-06-13T00:00:00.000Z',
        updatedAt: '2018-06-13T00:00:00.000Z'
      },
      {
        id: 20004,
        name: 'React',
        color: 'green',
        createdAt: '2018-06-13T00:00:00.000Z',
        updatedAt: '2018-06-13T00:00:00.000Z'
      },
      {
        id: 20005,
        name: 'Node.js',
        color: 'blue',
        createdAt: '2018-06-13T00:00:00.000Z',
        updatedAt: '2018-06-13T00:00:00.000Z'
      },
      {
        id: 20006,
        name: 'Java',
        color: 'orange',
        createdAt: '2018-06-13T00:00:00.000Z',
        updatedAt: '2018-06-13T00:00:00.000Z'
      },
      {
        id: 20007,
        name: 'FORTRAN',
        color: 'pink',
        createdAt: '2018-06-13T00:00:00.000Z',
        updatedAt: '2018-06-13T00:00:00.000Z'
      },
      {
        id: 20008,
        name: 'DROPPED',
        color: 'grey',
        createdAt: '2018-06-13T00:00:00.000Z',
        updatedAt: '2018-06-13T00:00:00.000Z'
      }
    ]
  }

  const coursePageLogic = {
    showDropdown: '',
    selectedTeacher: '',
    filterByAssistant: 0,
    filterByTag: []
  }

  const loading = {
    loading: false,
    loadingHooks: [],
    redirect: false,
    redirectHooks: [],
    redirectFailure: false
  }

  let mockFn = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <MassEmailPage
        courseData={coursePage}
        getOneCI={mockFn}
        coursePageInformation={mockFn}
        associateTeacherToStudent={mockFn}
        selectedInstance={coursePage}
        coursePageLogic={coursePageLogic}
        getAllTags={mockFn}
        courseReset={mockFn}
        tags={tags}
        loading={loading}
        resetLoading={mockFn}
        courseId={''}
        user={{ }}
        studentInstance={{ }}
        teacherInstance={{ }}
        courseInstance={{ }}
        coursePageReset={mockFn}
        sendMassEmail={mockFn}
        addRedirectHook={mockFn}
      />
    )
  })

  describe('MassEmailPage Component', () => {
    it('is ok', () => {
      true
    })

    // it('should render without throwing an error', () => {
    //   expect(wrapper.find('.CoursePage').exists()).toEqual(true)
    // })

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('renders view correctly', () => {
      expect(wrapper.find('.TeacherMassEmailPart').length).toEqual(1)
    })

    // it('assistant dropdown menu is not shown when page loads', () => {
    //   expect(wrapper.find('.AssistantDropdown').length).toEqual(0)
    // })
  })
})
