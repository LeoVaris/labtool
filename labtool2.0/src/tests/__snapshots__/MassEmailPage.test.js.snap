// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MassEmailPage /> MassEmailPage Component should render correctly 1`] = `
<div
  style={
    Object {
      "overflow": "auto",
    }
  }
>
  <div
    className="TeacherMassEmailPart"
  >
    <Header
      as="h2"
    >
      Send email to students 
    </Header>
    <div
      style={
        Object {
          "textAlign": "left",
        }
      }
    >
      <span>
        Filter by instructor 
      </span>
      <Dropdown
        additionLabel="Add "
        additionPosition="top"
        closeOnBlur={true}
        deburr={false}
        defaultValue={0}
        fluid={true}
        icon="dropdown"
        minCharacters={1}
        noResultsMessage="No results found."
        onChange={[Function]}
        openOnFocus={true}
        options={Array []}
        placeholder="Select Teacher"
        renderLabel={[Function]}
        searchInput="text"
        selectOnBlur={true}
        selectOnNavigation={true}
        selection={true}
        style={
          Object {
            "display": "inline",
          }
        }
        wrapSelection={true}
      />
      <span>
         Tag filters: 
      </span>
      <span>
        <Label>
          none
        </Label>
      </span>
    </div>
    <Form
      as="form"
      onSubmit={[Function]}
    >
      <HorizontalScrollable>
        <Table
          as="table"
          celled={true}
          compact={true}
          style={
            Object {
              "overflowX": "visible",
            }
          }
          unstackable={true}
        >
          <TableHeader
            as="thead"
          >
            <TableRow
              as="tr"
              cellAs="td"
            >
              <TableHeaderCell
                as="th"
                key="-1"
              >
                Send?
              </TableHeaderCell>
              <TableHeaderCell
                as="th"
              >
                Student
              </TableHeaderCell>
              <TableHeaderCell
                as="th"
              >
                id
              </TableHeaderCell>
              <TableHeaderCell
                as="th"
              >
                email
              </TableHeaderCell>
              <TableHeaderCell
                as="th"
              >
                Project Info
              </TableHeaderCell>
              <TableHeaderCell
                as="th"
                width="six"
              >
                 Instructor 
              </TableHeaderCell>
            </TableRow>
          </TableHeader>
          <TableBody
            as="tbody"
          >
            <TableRow
              as="tr"
              cellAs="td"
              className="TableRowForActiveStudents"
              key="10012"
            >
              <TableCell
                as="td"
              >
                <FormCheckbox
                  as={[Function]}
                  control={[Function]}
                  name="send10012"
                />
              </TableCell>
              <TableCell
                as="td"
              >
                Johan Wilhelm
                 
                Studerande
              </TableCell>
              <TableCell
                as="td"
              >
                <span>
                  014553242
                </span>
              </TableCell>
              <TableCell
                as="td"
              >
                <a
                  href="mailto:johan.studerande@helsinki.fi"
                >
                  johan.studerande@helsinki.fi
                </a>
              </TableCell>
              <TableCell
                as="td"
              >
                <span>
                  Tiran toinen labraprojekti
                  <br />
                  <a
                    href="http://github.com/tiralabra2"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    http://github.com/tiralabra2
                  </a>
                  <div
                    key="20001"
                  >
                    <Button
                      as="button"
                      className="mini ui red button"
                      compact={true}
                      floated="left"
                      onClick={[Function]}
                      role="button"
                    >
                      Javascript
                    </Button>
                  </div>
                </span>
              </TableCell>
              <TableCell
                as="td"
              >
                <span>
                  not assigned
                </span>
              </TableCell>
            </TableRow>
            <TableRow
              as="tr"
              cellAs="td"
              className="TableRowForActiveStudents"
              key="10031"
            >
              <TableCell
                as="td"
              >
                <FormCheckbox
                  as={[Function]}
                  control={[Function]}
                  name="send10031"
                />
              </TableCell>
              <TableCell
                as="td"
              >
                Teräs
                 
                Henkilö
              </TableCell>
              <TableCell
                as="td"
              >
                <span>
                  014666666
                </span>
              </TableCell>
              <TableCell
                as="td"
              >
                <a
                  href="mailto:teras.henkilo@helsinki.fi"
                >
                  teras.henkilo@helsinki.fi
                </a>
              </TableCell>
              <TableCell
                as="td"
              >
                <span>
                  Tira super projekti
                  <br />
                  <a
                    href="http://github.com/superprojekti"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    http://github.com/superprojekti
                  </a>
                  <div
                    key="20008"
                  >
                    <Button
                      as="button"
                      className="mini ui grey button"
                      compact={true}
                      floated="left"
                      onClick={[Function]}
                      role="button"
                    >
                      DROPPED
                    </Button>
                  </div>
                  <div
                    key="20002"
                  >
                    <Button
                      as="button"
                      className="mini ui yellow button"
                      compact={true}
                      floated="left"
                      onClick={[Function]}
                      role="button"
                    >
                      HTML
                    </Button>
                  </div>
                </span>
              </TableCell>
              <TableCell
                as="td"
              >
                <span>
                  not assigned
                </span>
              </TableCell>
            </TableRow>
            <TableRow
              as="tr"
              cellAs="td"
              className="TableRowForActiveStudents"
              key="10011"
            >
              <TableCell
                as="td"
              >
                <FormCheckbox
                  as={[Function]}
                  control={[Function]}
                  name="send10011"
                />
              </TableCell>
              <TableCell
                as="td"
              >
                Maarit Mirja
                 
                Opiskelija
              </TableCell>
              <TableCell
                as="td"
              >
                <span>
                  014578343
                </span>
              </TableCell>
              <TableCell
                as="td"
              >
                <a
                  href="mailto:maarit.opiskelija@helsinki.fi"
                >
                  maarit.opiskelija@helsinki.fi
                </a>
              </TableCell>
              <TableCell
                as="td"
              >
                <span>
                  Tiran labraprojekti
                  <br />
                  <a
                    href="http://github.com/tiralabra1"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    http://github.com/tiralabra1
                  </a>
                </span>
              </TableCell>
              <TableCell
                as="td"
              >
                <span>
                  not assigned
                </span>
              </TableCell>
            </TableRow>
          </TableBody>
        </Table>
      </HorizontalScrollable>
      <br />
      <FormTextArea
        as={[Function]}
        control={[Function]}
        defaultValue=""
        name="content"
        placeholder="Type email here..."
        required={true}
      />
      <Button
        as="button"
        content="Send"
        role="button"
        type="submit"
      />
      <br />
      <br />
      <Link
        replace={false}
        to="/labtool/courses/undefined"
      >
        <Button
          as="button"
          className="ui button"
          role="button"
          type="cancel"
        >
          Cancel
        </Button>
      </Link>
    </Form>
    <br />
  </div>
</div>
`;
